src/clients/SpokePoolClient/SpokePoolClient.ts

```diff
@@ -813,7 +813,10 @@ export abstract class SpokePoolClient extends BaseAbstractClient {
    * @param blockTag The block at which to query the fill status.
    * @returns The fill status for the given relay data.
    */
-  public abstract relayFillStatus(relayData: RelayData, blockTag?: number | "latest"): Promise<FillStatus>;
+  public abstract relayFillStatus(
+    relayData: RelayData,
+    blockTag?: number | "latest" | "confirmed"
```

Comment:
Exposing `confirmed` at this interface means that the caller needs to be aware of which chain type they're on, and we then need to validate in each chain-specific implementation that the correct tag was passed in. If we modify the type to be just an optional number then we could infer "chain-specific latest" if the input is undefined. That seems more platform-agnostic, though it means that we internally must decide on `processed` or `confirmed` for SVM. Based on @james-a-morris feedback about the delta being only 1 - 2 slots, it seems like we could probably live with `confirmed` for now. wdyt?

```suggestion
    blockTag?: number
```

Reply:
I like this idea! Fwiw, we aren't relying too much on this option, the only place where I see we are using "latest" explicitly is in [this line](https://github.com/across-protocol/relayer/blob/master/src/dataworker/Dataworker.ts#L1234) of the Dataworker. So I think this makes sense, wdyt @james-a-morris?

Reply:
This is how it would look like [d3cfa15](https://github.com/across-protocol/sdk/pull/990/commits/d3cfa155f4dfe77ace924d00ca560123ee59cfdc)

I wonder if we should change the name of this param too, blockTag might be confusing when used from the SVM Spoke Pool. Wdyt? It could be "atHeight" or something similar.